lib/app/features/tokenized_communities/providers/fat_address_data_provider.r.dart

```diff
@@ -120,5 +135,54 @@ Future<FatAddressData> _buildContentFatAddressData(
     externalTypePrefix: externalTypePrefix,
     creatorAddress: creatorAddress,
     creatorTokenAddress: creatorTokenAddress,
+    affiliateAddress: affiliateAddress,
   );
 }
+
+Future<String?> _resolveAffiliateAddress(
+  Ref ref, {
+  required String creatorMasterPubkey,
+  required String bscNetworkId,
+}) async {
+  try {
+    final ionIdentityClient = await ref.watch(ionIdentityClientProvider.future);
+    final socialProfile = await ionIdentityClient.users.getUserSocialProfile(
+      userIdOrMasterKey: creatorMasterPubkey,
+    );
+
+    final referralNickname = socialProfile.referral?.trim();
+    if (referralNickname == null || referralNickname.isEmpty) {
+      return null;
+    }
+
+    final referredUsers = await ionIdentityClient.users.searchForUsersByKeyword(
+      keyword: referralNickname,
+      searchType: SearchUsersSocialProfileType.startsWith,
+      limit: 20,
+      offset: 0,
+    );
+
+    final referralLower = referralNickname.toLowerCase();
+    final referredUser = referredUsers.firstWhere(
```

Comment:
it might be `firstWhereOrNull` to avoid that fake
```
const IdentityUserInfo(
        masterPubKey: '',
        ionConnectRelays: [],
      ),
    )
```